# Type: refactor
# Subtype: R10
# English Summary: Remove extractPsiFromValue() method
# Chinese Summary: 移除 extractPsiFromValue() 方法
# Entry Number: 24

# === Git Diff Content ===

diff --git a/plugins/groovy/src/org/jetbrains/plugins/groovy/mvc/projectView/AbstractMvcPsiNodeDescriptor.java b/plugins/groovy/src/org/jetbrains/plugins/groovy/mvc/projectView/AbstractMvcPsiNodeDescriptor.java
index d378f68..cee1d3f 100644
--- a/plugins/groovy/src/org/jetbrains/plugins/groovy/mvc/projectView/AbstractMvcPsiNodeDescriptor.java
+++ b/plugins/groovy/src/org/jetbrains/plugins/groovy/mvc/projectView/AbstractMvcPsiNodeDescriptor.java
@@ -17,7 +17,7 @@ package org.jetbrains.plugins.groovy.mvc.projectView;
 
 import com.intellij.ide.projectView.PresentationData;
 import com.intellij.ide.projectView.ViewSettings;
-import com.intellij.ide.projectView.impl.nodes.AbstractPsiBasedNode;
+import com.intellij.ide.projectView.impl.nodes.BasePsiNode;
 import com.intellij.navigation.ItemPresentation;
 import com.intellij.navigation.NavigationItem;
 import com.intellij.openapi.module.Module;
@@ -34,7 +34,7 @@ import org.jetbrains.annotations.Nullable;
 /**
  * @author Dmitry Krasilschikov
  */
-public abstract class AbstractMvcPsiNodeDescriptor extends AbstractPsiBasedNode<PsiElement> {
+public abstract class AbstractMvcPsiNodeDescriptor extends BasePsiNode<PsiElement> {
   public static final int FOLDER = 100;
   public static final int FILE = 110;
   public static final int CLASS = 5;
@@ -74,12 +74,6 @@ public abstract class AbstractMvcPsiNodeDescriptor extends AbstractPsiBasedNode<
     return super.contains(file);
   }
 
-  @Override
-  @Nullable
-  protected PsiElement extractPsiFromValue() {
-    return getValue();
-  }
-
   @Nullable
   @Override
   public String toTestString(@Nullable Queryable.PrintInfo printInfo) {
